apps/examples/wifi_manager_sample/wm_test_interop.c

```diff
@@ -0,0 +1,664 @@
+	}
+
+	WIO_TEST_SIGNAL;
+}
+
+static void wm_sta_disconnected(wifi_manager_disconnect_e disconn)
+{
+	WT_LOG(TAG, "disconnect res(%d)", disconn);
+	g_connect_status = 0;
+	WIO_TEST_SIGNAL;
+}
+
+static void wm_scan_done(wifi_manager_scan_info_s **scan_result, wifi_manager_scan_result_e res)
+{
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted soon eventually as you exit this function.
+	 */
+	if (scan_result == NULL) {
+		WIO_TEST_SIGNAL;
+		return;
+	}
+	wifi_manager_scan_info_s *wifi_scan_iter = *scan_result;
+	while (wifi_scan_iter != NULL) {
+		WT_LOG(TAG, "WiFi AP SSID: %-25s, BSSID: %-20s, Rssi: %d, Auth: %d, Crypto: %d",
+				wifi_scan_iter->ssid, wifi_scan_iter->bssid, wifi_scan_iter->rssi,
+				wifi_scan_iter->ap_auth_type, wifi_scan_iter->ap_crypto_type);
+
+		interop_ap_info_s *ap_info = (g_apconfig_list.ap_info + g_current_ap_idx);
+
+		//check AP
+		if (strncmp(wifi_scan_iter->ssid, ap_info->ap_config.ssid, WIFIMGR_SSID_LEN) == 0) {
+			WT_LOG(TAG, "WiFi AP SSID : [%-25s] found", wifi_scan_iter->ssid);
+			g_ap_found_status = 1;
+			break;
+		}
+
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+
+	WIO_TEST_SIGNAL;
+}
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	wm_sta_connected,
+	wm_sta_disconnected,
+	NULL, NULL,
+	wm_scan_done,
+};
+
+static void wm_interop_disconnect(void)
+{
+	WT_ENTER;
+
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+
+	/* Disconnect AP */
+	res = wifi_manager_disconnect_ap();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "disconnect fail (%d)", res);
+		return;
+	}
+	WIO_TEST_WAIT;
+
+	WT_LEAVE;
+}
+
+static void wm_interop_connect(wifi_manager_ap_config_s *apconfig)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+
+	print_wifi_ap_profile(apconfig);
+
+	res = wifi_manager_connect_ap(apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "AP connect failed");
+		return;
+	}
+	/* Wait for DHCP connection */
+	WT_LOG(TAG, " wait join done");
+	WIO_TEST_WAIT;
+
+	WT_LEAVE;
+}
+
+static void wm_interop_scan(wifi_manager_ap_config_s *ap_info)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	wifi_manager_scan_config_s sconfig;
+
+	sconfig.channel = 0;
+	sconfig.ssid_length = strlen(ap_info->ssid);
+	strncpy(sconfig.ssid, ap_info->ssid, sconfig.ssid_length + 1);
+
+	res = wifi_manager_scan_ap(&sconfig);
+
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "scan Fail");
+		return;
+	}
+	WIO_TEST_WAIT; // wait the scan result
+
+	WT_LEAVE;
+}
+
+static int wm_interop_store_file(char *buf, unsigned int buf_size, char *file_name)
+{
+	FILE *fp;
+
+	WT_ENTER;
+
+	fp = fopen(file_name, "a+");
+
+	if (!fp) {
+		WT_LOGE(TAG, "file open error(%d)", errno);
+		return -1;
+	}
+
+	int ret = fputs(buf, fp);
+	if (ret < 0) {
+		WT_LOGE(TAG, "file write error(%d)", errno);
+		fclose(fp);
+		return -1;
+	}
+
+	fclose(fp);
+
+	WT_LEAVE;
+	return 0;
+}
+
+static int wm_interop_save_ap_config(struct wt_options *ap_info)
+{
```

Comment:
I think we don't need feature, because TizenRT already provides similar features.
For example please check romfs. it'll add file to /rom folder from your host pc.
Or simple_file_transfer.py in tools/net/ it'll send your file to TizenRT through network.

Reply:
Since our Winodws and Ubuntu machines does not have wifi card, so it is convenient utility to add ap list in file.